rtgui/edit.h

```diff
@@ -152,6 +145,59 @@ class EditBuffer;
  *
  */
 
+
+
+class ObjectMOBuffer
+{
+private:
+
+    // Used to draw the objects where the color correspond to the object's ID, in order to find the correct object when hovering
+    Cairo::RefPtr<Cairo::ImageSurface> objectMap;
+    // If more than 254 objects has to be handled, objectMap2 contains the "upper part" of the 16 bit int value. objectMap2 will be NULL otherwise.
+    Cairo::RefPtr<Cairo::ImageSurface> objectMap2;
+    ObjectMode objectMode;
+
+protected:
+
+    // To avoid duplicated information, we points to a EditDataProvider that contains the current EditSubscriber
+    // instead of pointing to the EditSubscriber directly
+    EditDataProvider* dataProvider;
+
+    void createBuffer(int width, int height);
+    void resize(int newWidth, int newHeight, EditSubscriber* newSubscriber);
+    void flush();
+    EditSubscriber *getEditSubscriber ();
+
+public:
+    ObjectMOBuffer(EditDataProvider *dataProvider);
```

Comment:
The problem is again that it is problematic as the intent is unclear and the code becomes hard to follow, e.g. with the implicit constructor something like
````
buffer = dataProvider;
````
with `buffer` being of type `ObjectMOBuffer` is legal and would implicitly create a new instance of `ObjectMOBuffer` and assign it to `buffer`. At least to me, it would be very much unclear if that was actually intended or rather is a typo.
Implicitly constructing objects with a complicated life cycle that containing other objects is usually a bad thing. Implicit constructors of course do make sense in other places, e.g. implicitly constructing a `std::string` from a `const char*`, but they are definitely problematic as a default.